Roles landing page + reorg (#980)
osohq/oso: docs/content/node/guides/roles/getting-started.md

---
title: Getting started
weight: 2
description: >
    Get started with Oso Roles
---

# Getting started

When managing access to resources within an application, it can be useful to

> group permissions into **roles**, and assign these roles to users. This is

Reviewer: comma splice
```suggestion
group permissions into **roles** and assign these roles to users. This is
```

> known as **Role-Based Access Control (RBAC).** The `Oso` core library

Reviewer: ```suggestion
known as **Role-Based Access Control (RBAC).** The Oso library
```

comes with built-in configuration for role-based access control.

In this guide, we'll walk through the basics of starting to use the
roles feature.

## Setting up the Oso Instance

First, we'll cover some of the basics of integrating Oso into your
application.

> The `oso.Oso` class is the entrypoint to using Oso in our application.

Reviewer: the `oso.Oso` notation makes more sense for Python than JS
```suggestion
The `Oso` class is the entrypoint to using Oso in our application.
```

We usually will have a global instance that is created
during application initialization and shared across requests. In a Flask
application, you may attach this instance to the global flask object, or the
current application if it needs to be accessed outside of the application

> initialization process.

Reviewer: Flask mentioned in Node.js version of the doc — we should update the Node doc to mirror the recent updates made to the Python doc

> In order to enable the built-in roles features, we 
> `oso.Oso.enable_roles` method:

Reviewer: ```suggestion
In order to enable the built-in roles features, we call Oso's
`enableRoles()` method:
```

```js
const { Oso } = require('oso');

const oso = new Oso()

> oso.enableRoles()

Reviewer: this has to happen after at least a minimum viable roles configuration policy has been loaded

Reviewer: also we have to `await` this — `enableRoles()` now returns a Promise after the recent validation work

```

### Loading our policy

Oso uses the Polar language to define authorization policies. An
authorization policy specifies what requests are allowed and what data a
user can access. The policy is stored in a Polar file, along with your
code.

> Load the policy with the `Oso.load_file` function.

Reviewer: ```suggestion
Load the policy with Oso's `loadFile()` function:
```

```js

> oso.loadFile("authorization.polar")

Reviewer: ```suggestion
await oso.loadFile("authorization.polar")
```
